web-js/src/components/useTodos_graphql.js

```
@@ -0,0 +1,164 @@
+  }, [realmApp.currentUser]);
+
+  return client;
+}
+
+export function useTodos() {
+  const realmApp = useRealmApp();
+  const graphql = useApolloClient();
+  const taskCollection = React.useMemo(() => {
+    const mdb = realmApp.currentUser.mongoClient("mongodb-atlas");
+    return mdb.db("todo").collection("Task");
+  }, [realmApp.currentUser]);
+  const [todos, setTodos] = React.useState([]);
+  
+  const [loading, setLoading] = React.useState(true);
+  React.useEffect(() => {
+    const fetchTodos = async () => {
+      const { data } = await graphql.query({
+        query: gql`
+          query FetchAllTasks {
+            tasks {
+              _id
+              _partition
+              isComplete
+              summary
+            }
+          }
+        `,
+      });
+      return data.tasks
+    };
+
+    const watchTodos = async () => {
+      for await (const change of taskCollection.watch({ filter: {} })) {
+        const getTodoIndex = (todos) =>
+          todos.findIndex(
+            (t) => String(t._id) === String(change.documentKey._id)
+          );
+        switch (change.operationType) {
+          case "insert": {
+            setTodos((t) => [...t, change.fullDocument]);
+            break;
+          }
+          case "update":
+          case "replace": {
+            setTodos((oldTodos) => {
+              const idx = getTodoIndex(oldTodos);
+              return [
+                ...oldTodos.slice(0, idx),
+                change.fullDocument,
+                ...oldTodos.slice(idx + 1),
+              ];
+            });
+            break;
+          }
+          case "delete": {
+            setTodos((oldTodos) => {
+              const idx = getTodoIndex(oldTodos);
+              return [...oldTodos.slice(0, idx), ...oldTodos.slice(idx + 1)];
+            });
+            break;
+          }
+          default: {
+            // change.operationType will always be one of the specified cases, so we should never hit this default
```

If you should never hit this default, perhaps you should just remove this blank default or have it throw an error?

---

Ah yeah it should throw an error - good catch